# generated by patch-package 6.4.7 on 2021-10-01 13:21:55
#
# command:
#   npx patch-package karma-mocha-reporter
#
# declared package:
#   karma-mocha-reporter: 2.2.5
#
diff --git a/node_modules/karma-mocha-reporter/index.js b/node_modules/karma-mocha-reporter/index.js
index de36248..069b569 100644
--- a/node_modules/karma-mocha-reporter/index.js
+++ b/node_modules/karma-mocha-reporter/index.js
@@ -20,6 +20,7 @@ var MochaReporter = function (baseReporterDecorator, formatError, config) {
     var firstRun = true;
     var isRunCompleted = false;
     var internalPrefix = '$%$';
+    var consoleMessages = []
 
     /**
      * Returns the text repeated n times.
@@ -424,6 +425,40 @@ var MochaReporter = function (baseReporterDecorator, formatError, config) {
         }
     }
 
+    var Levels = ["error", "warn", "info", "debug", "log"];
+    function shortenMessage(message) {
+        var truncated = message.length > 200 ? `${message.substring(0, 200)}...` : message
+        return truncated.replace(/[\r\n]+/g, '')
+    }
+    function printCleanConsoleMessages(consoleMessages) {
+        var messagesFiltered = []
+        consoleMessages.forEach(m => {
+            const shortenedMessage = shortenMessage(m.log)
+            const alreadyAdded = messagesFiltered.findIndex(g => g.log === shortenedMessage)
+            if (alreadyAdded > -1) {
+                messagesFiltered[alreadyAdded].count += 1
+            } else {
+                messagesFiltered.push({type: m.type, log: shortenedMessage, count: 1})
+            }
+        })
+        var messagesGrouped = {}
+        Levels.forEach(level => {
+            messagesGrouped[level] = []
+        })
+        messagesFiltered.forEach(m => {
+            messagesGrouped[m.type].push(m)
+            if (!Levels.includes(m.type)) {
+                Levels.push(m.type)
+            }
+        })
+        Levels.forEach(l => {
+            messagesGrouped[l].forEach(m => {
+                self.write(`\n${l.toUpperCase()} (${m.count}): ${m.log}`)
+            })
+        })
+        
+    }
+
     /**
      * Returns a singularized or plularized noun for "test" based on test count
      *
@@ -553,6 +588,13 @@ var MochaReporter = function (baseReporterDecorator, formatError, config) {
         self._browsers.push(browser);
     };
 
+    self.onBrowserLog = function(browser, log, type) {
+        consoleMessages.push({
+            type: type,
+            log: log
+        })
+    }
+
     self.onRunComplete = function (browsers, results) {
         browsers.forEach(function (browser) {
             self.totalTime += browser.lastResult.totalTime;
@@ -601,6 +643,8 @@ var MochaReporter = function (baseReporterDecorator, formatError, config) {
         if (outputMode === 'autowatch') {
             outputMode = 'minimal';
         }
+
+        printCleanConsoleMessages(consoleMessages)
     };
 };
 
